src/test/java/wolox/training/controllers/BookTest.java

```diff
@@ -0,0 +1,174 @@
+package wolox.training.controllers;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+import java.util.ArrayList;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.any;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import static org.mockito.BDDMockito.given;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(BookController.class)
+class BookControllerIntegrationTest {
+
+  @Autowired
+  private MockMvc mvc;
+
+  @MockBean
+  private BookRepository bookRepository;
+
+  // write test cases here
+
+  private Book testBook = new Book("Doyle","image","title","subtitle","publisher","1999","500","isbn","terror");
+  private Book testBook2 = new Book("Doyle2","image2","title2","subtitle2","publisher2","1998","800","isbn2","fiction");
+  private JSONObject jsonBook;
+
+  {
+    try {
+      jsonBook = new JSONObject()
+          .put("author", testBook.getAuthor())
+          .put("image", testBook.getImage())
+          .put("title", testBook.getTitle())
+          .put("subtitle",testBook.getSubtitle())
+          .put("publisher", testBook.getPublisher())
+          .put("year", testBook.getYear())
+          .put("pages", testBook.getPages())
+          .put("isbn", testBook.getIsbn())
+          .put("genre", testBook.getGenre());
+    } catch (JSONException e) {
+      e.printStackTrace();
+    }
+  }
```

Comment:
Use @Before for this setup

Check this https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall